src/freenet/clients/http/DiagnosticToadlet.java

```diff
@@ -411,18 +413,54 @@ public int compare(PeerNodeStatus firstNode, PeerNodeStatus secondNode) {
 
 		// drawThreadPriorityStatsBox
 		textBuilder.append("Threads:\n");
-		int[] activeThreadsByPriority = stats.getActiveThreadsByPriority();
-		int[] waitingThreadsByPriority = stats.getWaitingThreadsByPriority();
-		for(int i=0; i<activeThreadsByPriority.length; i++) {
-			textBuilder.append(l10n("running")).append(": ").append(String.valueOf(activeThreadsByPriority[i])).append(" (").append(String.valueOf(i+1)).append(")\n");
-			textBuilder.append(l10n("waiting")).append(": ").append(String.valueOf(waitingThreadsByPriority[i])).append(" (").append(String.valueOf(i+1)).append(")\n");
-		}
+		textBuilder.append(threadsStats());
+
 		textBuilder.append("\n");
 		}
 
 		this.writeTextReply(ctx, 200, "OK", textBuilder.toString());
 	}
 
+	private StringBuilder threadsStats() {
+		StringBuilder sb = new StringBuilder();
+		// ID, Name, Priority, Group (system, main), Status, % CPU
+		sb.append(
+			String.format(
+				"%5s %-60s %5s %10s %-20s %-5s%n",
+				"ID",
+				"Name",
+				"Prio.",
+				"Group",
+				"Status",
+				"% CPU"
+			)
+		);
+
+		ThreadDiagnostics threadDiagnostics = node
+			.getNodeDiagnostics()
+			.getThreadDiagnostics();
+
+		List<NodeThreadInfo> threads = threadDiagnostics.getThreads();
+		threads.sort(
+			(o1, o2) -> Double.compare(o2.getCpuTime(), o1.getCpuTime())
```

Comment:
Is sorting in increasing order a good idea for a default? I’d argue that you want to see thread that are using lots of CPU as the first ones…

Reply:
That's the case yes, higher CPU % are listed at the top.

Reply:
What about replacing it with this? It makes the intent clearer and has less code written by us. 🙂 
```suggest
threads.sort(Comparator.comparing(NodeThreadInfo::getCpuTime).reversed());
```

Reply:
Less code is always better :)